File: leecode/js/066-Plus One.js
var plusOne = function(digits) {
  // 判斷相加後是否需進位
  var carry = 0;

  for(var i = digits.length - 1 ; i >= 0 ; i--){
      // 目前位數 = 目前位數+前面是否進位
      digits[i]= digits[i] + carry;

      // list最後一個數字，也就是個位數，給他+1
      if(i == digits.length - 1 ){
        digits[i] = digits[i] + 1;
      } 

      // 如果目前這個位數等於10(因為只+1，所以不會超過10)，進位
      if(digits[i] == 10){
        digits[i] = 0;
        carry = 1;
      } else {
        carry = 0;
      }
  }

  // 最後如果有進位
  if(carry == 1){
      digits.unshift(carry); 
  }

  return digits;
};

File: leecode/ruby/066-Plus One.rb
def plus_one(digits)
  digits =( digits.join.to_i + 1).to_s
  arr=Array.new 
  digits.size.times do |i|
     arr.push( digits[i].to_i  )  
  end 
  return arr
end

a = [1,2,3]

p plus_one(a)

# def plus_one(digits)
#   str= ""
#   digits.each do |index|
#     str = str + index.to_s
#   end 
#    data=str.to_i+1
#   newstr=data.to_s
#    arr=Array.new 
#   newstr.size.times do |i|
#      arr.push( newstr[i].to_i  )  
#   end 
#   return arr
# end